src/api.js

```diff
@@ -166,7 +172,19 @@ const Api = (options = {}) => {
         rb,
         hdl: [
           () => parse(request, params, event),
-          paramsOut => handler(paramsOut, context, rb, event)
+          async (paramsOut) => {
+            const result = await handler(paramsOut, context, rb, event);
+            if (rawAutoPruneFieldsParam !== undefined && paramsOut[rawAutoPruneFieldsParam.name] !== undefined) {
+              assert(get(result, 'isJsonResponse') === true, 'Can only auto prune JsonResponse.');
+              const autoPrunePath = rawAutoPruneFieldsParam.autoPrunePath;
+              objectRewrite({
+                retain: objectPaths.split(paramsOut[rawAutoPruneFieldsParam.name])
+              })(autoPrunePath !== null
+                ? get(result.payload, rawAutoPruneFieldsParam.autoPrunePath)
```

Comment:
why not reuse?

Reply:
lol that's the tiredness I was talking about